analytics.js

```diff
@@ -1337,10 +1337,20 @@ const eventsPushedAlready =
   window.rudderanalytics.push == Array.prototype.push;
 
 const argumentsArray = window.rudderanalytics;
-
-while (argumentsArray && argumentsArray[0] && argumentsArray[0][0] !== "load") {
-  argumentsArray.shift();
+/**
+ * Iterate the call stack until we find load call and
+ * then shift it to the beginning.
+ *
+ * Ex: Say the call stack currently have [page, identify, load, track]
+ * It will become [load, page, identify, track]
+ */
+for (let i = 0; i < argumentsArray.length; i++) {
```

Comment:
Use a while loop here as this array can be simultaneously updated by the client code.